src/hotspot/share/prims/jvmtiEnv.cpp

```diff
@@ -942,24 +942,18 @@ jvmtiError
 JvmtiEnv::SuspendThread(JavaThread* java_thread) {
   // don't allow hidden thread suspend request.
   if (java_thread->is_hidden_from_external_view()) {
-    return (JVMTI_ERROR_NONE);
+    return JVMTI_ERROR_NONE;
   }
-
-  {
-    MutexLocker ml(java_thread->SR_lock(), Mutex::_no_safepoint_check_flag);
-    if (java_thread->is_external_suspend()) {
-      // don't allow nested external suspend requests.
-      return (JVMTI_ERROR_THREAD_SUSPENDED);
-    }
-    if (java_thread->is_exiting()) { // thread is in the process of exiting
-      return (JVMTI_ERROR_THREAD_NOT_ALIVE);
-    }
-    java_thread->set_external_suspend();
+  if (java_thread->is_suspended()) {
+    return JVMTI_ERROR_THREAD_SUSPENDED;
   }
-
   if (!JvmtiSuspendControl::suspend(java_thread)) {
-    // the thread was in the process of exiting
-    return (JVMTI_ERROR_THREAD_NOT_ALIVE);
+    // Either the thread is already suspended or
+    // the thread was in the process of exiting:
```

Comment:
Nit: replace this line with: "// it was in process of exiting."

Reply:
Fixed